.github/workflows/release.yml

```diff
@@ -85,7 +84,65 @@ jobs:
         -DTI_BUILD_TESTS:BOOL=ON
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+        with:
```

Comment:
a lot of extra code here. Could you explain/comment what this is doing, relative to the existing scripts?